### DISCUSSION TITLE
About the quaternion convention

### DISCUSSION THREAD

**Discussion by simon-bachhuber** [google/brax#337] [Q&A] [answered]
Suppose one starts with the docstring from `base.Geometry`

```
    transform: transform from this geometry's coordinate space to its parent
      link, or to world space in the case of unparented geometry
``` 

And now we use this transform to rotate a vector
```
r_parent = math.rotate(r_geom, transform.rot)
```

What is the interpretation here? Is
- `r_parent` the old vector `r_geom` but in basis of `parent`
- or is `r_parent` a new rotated vector in basis of `geom` frame

Note: These two interpretations are related by inverting `transform.rot`

**Comment by btaba** [comment, marked as answer]
geometry.transform is the position/orientation of the geom in the parent frame, that docstring is a bit confusing
Here's an example of usage:
https://github.com/google/brax/blob/ff3ff641097699703087e1dc0a7b6e8305d78270/brax/geometry/contact.py#L499

so in your case, r_parent is r_geom rotated by transform.rot in the parent frame
